flax/linen/module.py

```diff
@@ -1104,6 +1104,17 @@ def other_fn(instance, ...):
     if method is None:
       method = self.__call__
     method = _get_unbound_fn(method)
+    # Try to detect if user accidentally passed {'params': {'params': ...} or
```

Comment:
This test could go in core/scope.py (apply method) to be more general.

Reply:
Done.